.github/workflows/wheels-dependencies.sh

```diff
@@ -1,42 +1,98 @@
+        echo "ERROR: Pillow builds only support a single architecture in CIBW_ARCHS."
         exit 1
     fi
+}
+
+# Setup that needs to be done before multibuild utils are invoked. Process
+# potential cross-build platforms before native platforms to ensure that we pick
+# up the cross environment.
+PROJECTDIR=$(pwd)
+if [[ "$CIBW_PLATFORM" == "ios" ]]; then
+    check_cibw_archs
+    # On iOS, CIBW_ARCHS is actually a multi-arch - arm64_iphoneos,
+    # arm64_iphonesimulator or x86_64_iphonesimulator. Split into the CPU
+    # platform, and the iOS SDK.
+    PLAT=$(echo $CIBW_ARCHS | sed "s/\(.*\)_\(.*\)/\1/")
+    IOS_SDK=$(echo $CIBW_ARCHS | sed "s/\(.*\)_\(.*\)/\2/")
+
+    # Build iOS builds in `build/iphoneos` or `build/iphonesimulator`
+    # (depending on the build target). Install them into `build/deps/iphoneos`
+    # or `build/deps/iphonesimulator`
+    WORKDIR=$(pwd)/build/$IOS_SDK
+    BUILD_PREFIX=$(pwd)/build/deps/$IOS_SDK
+    PATCH_DIR=$(pwd)/patches/iOS
+
+    # GNU tooling insists on using aarch64 rather than arm64
+    if [[ $PLAT == "arm64" ]]; then
+        GNU_ARCH=aarch64
+    else
+        GNU_ARCH=x86_64
+    fi
+
+    IOS_SDK_PATH=$(xcrun --sdk $IOS_SDK --show-sdk-path)
+    CMAKE_SYSTEM_NAME=iOS
+    IOS_HOST_TRIPLE=$PLAT-apple-ios$IPHONEOS_DEPLOYMENT_TARGET
+    if [[ "$IOS_SDK" == "iphonesimulator" ]]; then
+        IOS_HOST_TRIPLE=$IOS_HOST_TRIPLE-simulator
+    fi
 
+    # GNU Autotools doesn't recognize the existence of arm64-apple-ios-simulator
+    # as a valid host. However, the only difference between arm64-apple-ios and
+    # arm64-apple-ios-simulator is the choice of sysroot, and that is
+    # coordinated by CC, CFLAGS etc. From the perspective of configure, the two
+    # platforms are identical, so we can use arm64-apple-ios consistently.
+    # This (mostly) avoids us needing to patch config.sub in dependency sources.
+    HOST_CONFIGURE_FLAGS="--disable-shared --enable-static --host=$GNU_ARCH-apple-ios --build=$GNU_ARCH-apple-darwin"
+
+    # CMake has native support for iOS. However, most of that support is based
+    # on using the Xcode builder, which isn't very helpful for most of Pillow's
+    # dependencies. Therefore, we lean on the OSX configurations, plus CC, CFLAGS
+    # etc. to ensure the right sysroot is selected.
+    HOST_CMAKE_FLAGS="-DCMAKE_SYSTEM_NAME=$CMAKE_SYSTEM_NAME -DCMAKE_SYSTEM_PROCESSOR=$GNU_ARCH -DCMAKE_OSX_DEPLOYMENT_TARGET=$IPHONEOS_DEPLOYMENT_TARGET -DCMAKE_OSX_SYSROOT=$IOS_SDK_PATH -DBUILD_SHARED_LIBS=NO"
+
+    # Meson needs to be pointed at a cross-platform configuration file
+    # This will be generated once CC etc. have been evaluated.
+    HOST_MESON_FLAGS="--cross-file $WORKDIR/meson-cross.txt -Dprefer_static=true -Ddefault_library=static"
+
+elif [[ "$(uname -s)" == "Darwin" ]]; then
+    check_cibw_archs
     # Build macOS dependencies in `build/darwin`
     # Install them into `build/deps/darwin`
+    PLAT="${CIBW_ARCHS:-$AUDITWHEEL_ARCH}"
```

Comment:
```suggestion
    PLAT=$CIBW_ARCHS
```
`CIBW_ARCHS` isn't empty in our macOS cibuildwheel jobs.